llvm/docs/DirectX/DXILResources.rst

```diff
@@ -0,0 +1,425 @@
+representations for the various resources, albeit with a few things
+parameterized. This is different than DXIL, as simplifying the types to
+something like "dx.srv" and "dx.uav" types would mean the operations on these
+types would have to be overly generic.
+
+Buffers
+-------
+
+.. code-block:: llvm
+
+   target("dx.TypedBuffer", ElementType, IsWriteable, IsROV)
+   target("dx.RawBuffer", ElementType, IsWriteable, IsROV)
+
+We need two separate buffer types to account for the differences between the
+16-byte `bufferLoad`_ / `bufferStore`_ operations that work on DXIL's
+TypedBuffers and the `rawBufferLoad`_ / `rawBufferStore`_ operations that are
+used for DXIL's RawBuffers and StructuredBuffers. We call the latter
+"RawBuffer" to match the naming of the operations, but it can represent both
+the Raw and Structured variants.
+
+For TypedBuffer, the element type must be an integer or floating point type.
+For RawBuffer the type can be an integer, floating point, or struct type.
+HLSL's ByteAddressBuffer is represented by an `i8` element type.
+
+These types are generally used by BufferLoad and BufferStore operations, as
+well as atomics.
+
+There are a few fields to describe variants of all of these types:
+
+.. list-table:: Buffer Fields
+   :header-rows: 1
+
+   * - Field
+     - Description
+   * - ElementType
+     - Type for a single element, such as ``i8``, ``v4f32``, or a structure
+       type.
+   * - IsWriteable
+     - Whether or not the field is writeable. This distinguishes SRVs (not
+       writeable) and UAVs (writeable).
+   * - IsROV
+     - Whether the UAV is a rasterizer ordered view. Always ``0`` for SRVs.
+
+.. _bufferLoad: https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/DXIL.rst#bufferload
+.. _bufferStore: https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/DXIL.rst#bufferstore
+.. _rawBufferLoad: https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/DXIL.rst#rawbufferload
+.. _rawBufferStore: https://github.com/microsoft/DirectXShaderCompiler/blob/main/docs/DXIL.rst#rawbufferstore
+
+Resource Operations
+===================
+
+.. _dxil-resources-handles:
+
+Resource Handles
+----------------
+
+We provide a few different ways to instantiate resources in the IR via the
+``llvm.dx.handle.*`` intrinsics. These intrinsics are overloaded on return
+type, returning an appropriate handle for the resource, and represent binding
+information in the arguments to the intrinsic.
+
+The three operations we need are ``llvm.dx.handle.fromBinding``,
+``llvm.dx.handle.fromHeap``, and ``llvm.dx.handle.fromPointer``. These are
+rougly equivalent to the DXIL operations ``dx.op.createHandleFromBinding``,
+``dx.op.createHandleFromHeap``, and ``dx.op.createHandleForLib``, but they fold
+the subsequent ``dx.op.annotateHandle`` operation in. Note that we don't have
+an analogue for `dx.op.createHandle`_, since ``dx.op.createHandleFromBinding``
+subsumes it.
```

Comment:
Is the intent to translate `llvm.dx.handle.FromBinding` to `dx.op.createHandle` directly for older shader models?

Reply:
Yes.